pkg/pg/txdb.go

```diff
@@ -0,0 +1,548 @@
+package pg
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/jmoiron/sqlx"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/utils"
+)
+
+// txdb is a simplified version of https://github.com/DATA-DOG/go-txdb
```

Comment:
Since you are introducing our own custom txdb here (which is a copy from chainlink), have you considered just using the DATA-DOG one (which is at least regularly updated)?

Reply:
cc @samsondav

Reply:
Its not a fork its a rewrite and its relatively small. Just put it here and import wherever you need it then its only needed once.

If it ain't broken don't fix it, and this ain't broke.

Reply:
We have so many tests in chainlink repo currently dependent on txdb, that I'm pretty sure migrating them to anything else would be a huge project. From what I understand, our txdb is based on DATA-DOG's txdb but customized toward chainlink's specific needs--but I haven't done a comparison of the two myself. (Although it's possible DATA-DOG supports more now than they did when ours was built?)

The context for the present PR is that we were looking for the fastest way to get db support including unit testing for Solana up and running so we can meet some very aggressive milestones toward getting CCIP up and running on Solana.  So we've stayed as closely as possible for now to the existing paradigm in order to avoid introducing unknown delay factors.  I have cleaned up our txdb however, and made some minor improvements such as better context handling.

Reply:
> and we will now do the same in the chainlink-common

Just to be clear, this is a move not a copy... the PR this is linked to in chainlink repo changes all of the imports of it to point here and removes it from that repo